server/src/main/java/org/apache/druid/catalog/model/ModelProperties.java

```diff
@@ -47,6 +49,7 @@
   interface PropertyDefn<T>
   {
     String name();
+    Map<String, Object> attributes();
```

Comment:
Would be good to update the javadoc for this class with information about what these attributes are